web/static/js/uploadManager.js

```diff
@@ -0,0 +1,50 @@
+// @flow
+import random from 'lodash/random'
+
+const uploads = {}
+
+export const upload = (file: Object, url: string, onCompleted: ((response: Object) => mixed), onProgress: ((total: number, loaded: number) => mixed), onAbort: (() => mixed), onError: (description: ?string) => mixed) => {
+  const formData = new FormData()
+  formData.append('file', file)
+  const xhr = new XMLHttpRequest()
+  const uploadId = random(100)
+
+  xhr.open('POST', url, true)
+
+  xhr.onreadystatechange = function() {
+    // state XMLHttpRequest.DONE == 4
+    if (this.readyState === 4 && this.status === 200) {
+      delete uploads[uploadId]
```

Comment:
Seems like the upload is removed from the dictionary only when it finishes successfully. I think it should be always removed when the request is completed, regardless of the result.